hazelcast/src/main/java/com/hazelcast/quorum/QuorumService.java

```diff
@@ -31,4 +31,19 @@
      * @throws NullPointerException     if quorumName is null
      */
     Quorum getQuorum(String quorumName) throws IllegalArgumentException;
+
+    /**
+     * Ensures that the quorum with the given name is present.
+     * Throws a QuorumException if quorum not present.
+     * Does not throw exception if quorumName null or quorum undefined.
+     *
+     * If (requiredQuorumPermissionType == READ) -> will check for presence of READ or READ_WRITE quorum<br/>
+     * If (requiredQuorumPermissionType == WRITE) -> will check for presence of WRITE or READ_WRITE quorum<br/>
+     * If (requiredQuorumPermissionType == READ_WRITE) -> will check for presence of READ_WRITE quorum<br/>
+     *
+     * @param quorumName                   quorum name to ensure, can be null or empty
+     * @param requiredQuorumPermissionType type of quorum required
+     * @throws QuorumException if quorum defined and not present
+     */
+    void ensureQuorumPresent(String quorumName, QuorumType requiredQuorumPermissionType) throws QuorumException;
```

Comment:
I believe this is no longer needed

Reply:
It's required for replicated map where everything is local: https://github.com/tombujok/hazelcast/blob/a0e8c3528b5ea482ceca31ee10dd3b59dee7ee5b/hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/ReplicatedMapProxy.java#L174

Reply:
Also Ringbuffer in 2 places: https://github.com/tombujok/hazelcast/blob/093be01c36ae9b83fbc332c323fadf689714f41a/hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferProxy.java#L89